pkg/kubelet/apis/cri/v1alpha1/runtime/constants.go

```diff
@@ -25,3 +25,23 @@ const (
 	// NetworkReady means the runtime network is up and ready to accept containers which require network.
 	NetworkReady = "NetworkReady"
 )
+
+// StreamType is the type of the stream in CRI container log.
+type StreamType string
+
+const (
+	// Stdout is the stream type for stdout.
+	Stdout StreamType = "stdout"
+	// Stderr is the stream type for stderr.
+	Stderr StreamType = "stderr"
+)
+
+// LogTag is the tag of a log line in CRI container log.
+type LogTag string
+
+const (
+	// Partial means the line is part of multiple lines.
+	Partial LogTag = "P"
```

Comment:
The last line could still be P, right?

Reply:
E.g. container could just result in
```
2016-10-06T00:17:09.669794202Z stdout P log content 1
```
And in the parser we get that as `log content 1` without a new line?

Reply:
If there is a newline at the end, or an io.EOF is hit without follow, yeah, the current logic will print that out in `kubectl logs`.

We could choose to discard it, current implementation is to keep it.

Reply:
Thanks! I verified it works for cri-o with a PR testing this. I prefer that we retain that so we don't lose output form a partial last line.